Topics/Software_Tools/Git/Automated_Deployment.md

```diff
@@ -0,0 +1,66 @@
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; The benefits of continuous delivery are that your team releases more often and deploying software is less complex since it is automated. Most importantly, it releases pressure on small changes, since they can always be quickly undone or fixed with future releases. However, it requires automated deployment, which takes time to set up, and the team needs to "embrace" flaging features so that incomplete features are not released. 
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The benefits of continuous deployment are faster development, less risky releases (since there are fewer changes per release), and constant improvements for customers. Issues with releases can be fixed more quickly, since development does not need to pause for releases. Releases are also not scheduled, so there is less pressure to finish everything by a certain deadline. However, just like delivery, feature flags **must** be a part of the coding culture, and application documentation must stay up to date with deployments. This is in order to ensure clarity on the included features in each release.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;However, most importantly, both continuous delivery and continuous deployment can only occur if the new changes pass all the tests from continuous integration. This means that in order to successfully implement continuous delivery or deployment, you need to have an extensive set of test for your code that can verify that the changes are suitable for these environments. Otherwise, continuous delivery or deployment can lead to issues within staging and testing environments, or, worse, issues for customers.
+ 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To summarize, automated deployment is used for continous deliver or deployment. In order for this to work, continuous integration needs to be used in order to extensively test the code and ensure it is acceptable to deploy to whichever environment (depending on if delivery or deployment is used). Although it requires a strong CI, it leads to faster development, less risky releases, and more customer feedback. 
+ <br>
+ <br>
+To learn more about CI/CD, please see the following article from Atlassian, an established software company from Australia: https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment
+ - To learn more about the difference between continous delivery and continuous deployment, see the following article: https://docs.github.com/en/actions/deployment/about-deployments/deploying-with-github-actions
```

Comment:
I think you may have included the wrong link here - you include the same one under the GitHub actions section. But I agree, a resource on the distinction between CI and CD here would be helpful.

also: replace 'continous' with 'continuous'

Reply:
Fixed